### PR TITLE
Update to Kotlin v2

### PR DESCRIPTION
**Pull request by pahill**
And a few small updates to support that release.

### CODE REVIEW

File: shared/build.gradle.kts [hunk 1]
```diff
@@ -1,15 +1,18 @@
+import org.jetbrains.kotlin.gradle.ExperimentalKotlinGradlePluginApi
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+
 plugins {
     alias(libs.plugins.android.library)
     alias(libs.plugins.kotlinx.serialization)
     alias(libs.plugins.kotlin.multiplatform)
+    alias(libs.plugins.compose.compiler) apply false
```
Comment (terrakok) [inline shared/build.gradle.kts:8]: why do we need the plugin here?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,21 +1,20 @@
 [versions]
-agp = "8.1.4"
-kotlin = "1.9.22"
+agp = "8.3.2"
```
Comment (terrakok) [inline gradle/libs.versions.toml:2]: why is not the latest?

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #Tue Apr 06 10:19:50 MSK 2021
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
```
Comment (terrakok) [inline gradle/wrapper/gradle-wrapper.properties:3]: the latest is 8.8

**Review by terrakok — CHANGES_REQUESTED** [review]

File: shared/build.gradle.kts (hunk 1 above)
Comment (pahill) [reply shared/build.gradle.kts]: Oops, wrong place.

File: gradle/libs.versions.toml (hunk 2 above)
Comment (pahill) [reply gradle/libs.versions.toml]: I thought it was compatible with AS, but looks like it's ok. Thank you

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (pahill) [reply gradle/wrapper/gradle-wrapper.properties]: @terrakok All updated, please check again when you have time. Thank you for your help!

**Pull request merged**